src/Microsoft.AspNet.OData.Shared/DeltaDeletedEntityObjectOfT.cs

```diff
@@ -0,0 +1,129 @@
+        /// </summary>
+        /// <param name="structuralType">The derived entity type or complex type for which the changes would be tracked.
+        /// <paramref name="structuralType"/> should be assignable to instances of <typeparamref name="TStructuralType"/>.
+        /// </param>
+        /// <param name="dynamicDictionaryPropertyInfo">The property info that is used as dictionary of dynamic
+        /// properties. <c>null</c> means this entity type is not open.</param>        
+        /// <param name="instanceAnnotationsPropertyInfo">The property info that is used as container for Instance Annotations</param>
+        public DeltaDeletedEntityObject(Type structuralType, PropertyInfo dynamicDictionaryPropertyInfo, PropertyInfo instanceAnnotationsPropertyInfo)
+            : this(structuralType, updatableProperties: null , dynamicDictionaryPropertyInfo, instanceAnnotationsPropertyInfo)
+        {
+ 
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="DeltaDeletedEntityObject{TStructuralType}"/>.
+        /// </summary>
+        /// <param name="structuralType">The derived entity type or complex type for which the changes would be tracked.
+        /// <paramref name="structuralType"/> should be assignable to instances of <typeparamref name="TStructuralType"/>.
+        /// </param>
+        /// <param name="updatableProperties"> Properties that can be updated</param>
+        /// <param name="dynamicDictionaryPropertyInfo">The property info that is used as dictionary of dynamic
+        /// properties. <c>null</c> means this entity type is not open.</param>
+        /// <param name="instanceAnnotationsPropertyInfo">The property info that is used as container for Instance Annotations</param>
+        public DeltaDeletedEntityObject(Type structuralType, IEnumerable<string> updatableProperties, PropertyInfo dynamicDictionaryPropertyInfo, PropertyInfo instanceAnnotationsPropertyInfo)
+            : this(structuralType, updatableProperties, dynamicDictionaryPropertyInfo, false, instanceAnnotationsPropertyInfo)
+        {
+            
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="DeltaDeletedEntityObject{TStructuralType}"/>.
+        /// </summary>
+        /// <param name="structuralType">The derived entity type or complex type for which the changes would be tracked.
+        /// <paramref name="structuralType"/> should be assignable to instances of <typeparamref name="TStructuralType"/>.
+        /// </param>
+        /// <param name="updatableProperties"> Properties that can be updated</param>
+        /// <param name="dynamicDictionaryPropertyInfo">The property info that is used as dictionary of dynamic
+        /// properties. <c>null</c> means this entity type is not open.</param>
+        /// <param name="isComplexType">To determine if the entity is a complex type</param>
```

Comment:
To keep the API a bit consistent, I'd make this the last argument instead of sandwiching it between `dynamicDictionaryPropertyInfo` and `instanceAnnotationsPropertyInfo`...